datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```
@@ -366,6 +446,9 @@ internal class QuestionnaireViewModel(application: Application, state: SavedStat
    * when first opening this questionnaire. Otherwise, returns `null`.
    */
   private fun Questionnaire.getInitialPagination(): QuestionnairePagination? {
+    detectCalculatedExpressionCyclicDependency()
```

Should `detectCalculatedExpressionCyclicDependency` be called  right after the `linkIdToQuestionnaireItemMap` is generated to fail early.